### ISSUE TITLE
Dependency updates for Kubernetes v1.23.x

### ISSUE THREAD

**Issue by shaneutt** [Kong/kubernetes-ingress-controller#2072]
### Problem Statement

Several recent dependency updates for a minor version bump aligning with Kubernetes `v1.23.x` were not backwards compatible:

- https://github.com/Kong/kubernetes-ingress-controller/pull/2060
- https://github.com/Kong/kubernetes-ingress-controller/pull/2061
- https://github.com/Kong/kubernetes-ingress-controller/pull/2062
- https://github.com/Kong/kubernetes-ingress-controller/pull/2063
- https://github.com/Kong/kubernetes-ingress-controller/pull/2064
- https://github.com/Kong/kubernetes-testing-framework/pull/173
- https://github.com/Kong/kubernetes-testing-framework/pull/174
- https://github.com/Kong/kubernetes-testing-framework/pull/175
- https://github.com/Kong/kubernetes-testing-framework/pull/176

As seen above the dependabot updates were not able to complete.

The purpose of the task is to make these updates manually and make the code adjustments needed to support the new versions.

- [ ] perform the dependency updates here in KIC
- [ ] ALSO perform the dependency updates in [KTF](https://github.com/kong/kubernetes-testing-framework)

**Comment by shaneutt** [comment]
There's a web of dependency changes and backwards incompatibilities here that boils down to two specific problems:

- `sigs.k8s.io/controller-runtime` compatibility is only available in the `main` branch at the moment, we need to wait for a release
- `knative.dev/serving` has not yet caught up with the upstream changes, not even in `main`, we need to wait for updates and a release

Since we don't want to be tagging for `main` or specific SHAs, this will have to be considered blocked on upstream for a time.

**Comment by rainest** [comment]
controller-runtime has released an updated version for both 0.23 and logr compatibility. https://github.com/Kong/kubernetes-ingress-controller/pull/2105 bumps to that and makes some minor changes to our code for compatibility. Knative definitely still broken:

```
Error: ../../../go/pkg/mod/knative.dev/serving@v0.28.0/pkg/apis/serving/fieldmask.go:312:5: out.Handler undefined (type *"k8s.io/api/core/v1".Probe has no field or method Handler)
Error: ../../../go/pkg/mod/knative.dev/serving@v0.28.0/pkg/apis/serving/fieldmask.go:312:18: in.Handler undefined (type *"k8s.io/api/core/v1".Probe has no field or method Handler)
Error: ../../../go/pkg/mod/knative.dev/serving@v0.28.0/pkg/apis/serving/fieldmask.go:325:22: undefined: "k8s.io/api/core/v1".Handler
Error: ../../../go/pkg/mod/knative.dev/serving@v0.28.0/pkg/apis/serving/fieldmask.go:329:13: undefined: "k8s.io/api/core/v1".Handler
Error: ../../../go/pkg/mod/knative.dev/serving@v0.28.0/pkg/apis/serving/k8s_validation.go:716:8: p.Handler undefined (type *"k8s.io/api/core/v1".Probe has no field or method Handler)
```

**Comment by rainest** [comment]
Knative has their 1.23 update scheduled a while out, for Oct 2022, but per [a conversation with their devs](https://knative.slack.com/archives/C93E33SN8/p1642461515040000) there are other ways to solve this: we don't use Knative libs outside tests, to create their objects. We can just use https://pkg.go.dev/k8s.io/client-go/dynamic to remove the dependency.

**Comment by Mason-Harding** [comment]
Hello.  This still seems to be an issue with the  knative.dev/serving@latest.  Is there something I need to do to use this library?  Thanks.

**Comment by shaneutt** [comment]
@Mason-Harding could you please file a new issue and let us know the specifics of what's not working?

**Comment by Mason-Harding** [comment]
Will do, thanks!
